addons/nightvision/functions/fnc_refreshGoggleType.sqf

```diff
@@ -60,6 +69,14 @@ if (alive ACE_player) then {
             TRACE_1("souce: binocular",binocular ACE_player); // Source is from player's binocular (Rangefinder/Vector21bNite)
             private _config = configFile >> "CfgWeapons" >> (binocular ACE_player);
             if (isNumber (_config >> QGVAR(generation))) then {_nvgGen = getNumber (_config >> QGVAR(generation));};
+            // Gets proper Params' Array from CfgWeapons (JDT & AleM)
+            if (isArray (_config >> "colorPreset")) then {_preset = getArray (_config >> "colorPreset");};
+            // Selection from Array (JDT & AleM)
+            _offset = _preset select 0;
+            _blend = _preset select 1;
+            _colorize = _preset select 2;
+            _weight = _preset select 3;
```

Comment:
Couldn't this be `_preset params ["_offset", "_blend", "_colorize", "_weight"]`?

Reply:
I think that would cause a scope issue

Reply:
Yes would cause a scope issue. But I don't even see the use of these variables. They are never used till line 115 and they are always the same as in the `_preset` variable.
Remove them all and use `params` on `_preset` on line 111

Reply:
Ok, dedmen, will do.